### PR TITLE
test: add HostHackathon.js coverage for validation, date-range, and auth guard

### LINKED ISSUE #10114: Zero test coverage on HostHackathon.js — form validation (including date-range logic) and submission flow untested
## Problem

`src/Pages/Hackathons/HostHackathon.js` (561 lines) — the hackathon-hosting form organizers use to submit a new hackathon — has zero behavioral test coverage. The only test referencing this file is `tests/hackathonAriaLabels.test.mjs`, which does a static regex scan of the raw source text for a generic `aria-label="button"` string; it does not exercise any form logic, validation, or submission behavior.

This form has real, non-trivial logic:

- `validateForm()` (lines 80–141): required-field checks (`hackathonName`, `organizerName`, `email`, `location`, `startDate`, `endDate`, `description`), length bounds on `hackathonName`/`organizerName`/`location` (3–100 chars) and `description` (20–2000 chars), an email regex, an optional website URL regex, a **date-range validation** (`startDate` cannot be in the past; `endDate` cannot be before `startDate`), and a `participantLimit` numeric floor (≥ 1).
- `handleSubmit()` (lines 143–215): re-validates on submit, scrolls to and focuses the first invalid field, sanitizes text fields via `sanitizeInputText`, calls `hostHackathon(...)` with a bearer token, and shows a success/error toast (note: unlike the sibling `SubmitProject.js` flow, there is **no retry action** on failure here — a good example of the kind of inconsistency this test gap lets go unnoticed).
- An auth guard (lines 20–25) redirecting unauthenticated users to `/login` on mount, re-checked again in `handleSubmit`.

None of this — the date-range logic, the length/regex validation, the auth guard, or the submit/error path — has any automated coverage today.

## Impact

A regression in any of the following would currently ship silently:
- The `endDate < startDate` check being broken, allowing a hackathon to be hosted with an invalid date range.
- The `startDate` "cannot be in the past" check silently passing through a past date.
- The auth guard failing to block an unauthenticated submission.
- The error toast losing its message or the form failing to reset on success.

## Steps to Reproduce

```
grep -rl "HostHackathon" tests src/__tests__ src/tests e2e
# only match: tests/hackathonAriaLabels.test.mjs (static aria-label
# regex scan of the raw source text — not a behavioral test)
```

## Expected Behaviour

`HostHackathon.js` should have test coverage matching its role as the entry point for every hackathon on the platform, comparable to the coverage auth and event registration already have (`tests/e2e/auth.spec.js`, `tests/e2e/event-registration-flow.spec.js`, etc.).

## Proposed Fix

Add tests covering, at minimum:

1. **`validateForm()` logic** (directly or via component interaction): required fields, length bounds on name/organizer/location/description, the email and website-URL regexes, and specifically the date-range rules (`startDate` in the past, `endDate` before `startDate`), and the `participantLimit` floor.
2. **Component tests** (React Testing Library) for:
   - Submitting with invalid/missing fields shows the expected errors and focuses the first invalid field.
   - A successful submission calls `hostHackathon` with sanitized data, shows the success toast, and resets the form.
   - A failed submission shows the error toast with the message from the thrown error.
3. **Auth guard**: an unauthenticated user is redirected to `/login` on mount and again if a submit is attempted while unauthenticated.

## Files Affected

- `src/Pages/Hackathons/HostHackathon.js` — component under test, no changes expected
- New test file(s) under `tests/` or `src/__tests__/`

## Related Issues

None — existing hackathon-related tests (`hackathonAriaLabels.test.mjs`, `hackathonFilterUtils.test.mjs`) cover accessibility labels and filter logic, not this form's validation/submission behavior. (Companion issue to #10113, which covers the same gap on `SubmitProject.js`.)

---
Complexity: Level 2 | Program: GSSoC '26

### PR DESCRIPTION
**Pull request by trimoyee-g**
## Summary
- `HostHackathon.js` (hackathon-hosting form) had no behavioral test coverage — the only match was `tests/hackathonAriaLabels.test.mjs`, a static regex scan of the raw source, not a behavioral test.
- Extracted `validateForm()` into `src/utils/hostHackathonValidation.js` (pure, no React deps) so it can be unit tested via `node --test`, matching this repo's dominant convention. Split into small single-purpose functions (required fields, length bounds, email/website regex, date-range, participant limit) with no compound (`&&`/`||`-stacked) conditionals and no nested `if`s, to avoid the Complex Method / Complex Conditional / Bumpy Road findings CodeScene raised on the companion `SubmitProject.js` PR (#10147). `HostHackathon.js` now imports and calls it; behavior is unchanged.
- Added `tests/hostHackathonValidation.test.mjs` covering required fields, length bounds (name/organizer/location/description), email regex, website URL regex, the date-range rules (`startDate` in the past, `endDate` before `startDate`), and the `participantLimit` floor.
- Added `src/Pages/Hackathons/HostHackathon.test.js` (React Testing Library) covering:
  - Auth guard: redirect to `/login` on mount when unauthenticated, and blocked+redirected submit attempt while unauthenticated.
  - Empty-submit error display, scroll-into-view + focus on the first invalid field.
  - Successful submission: sanitized payload sent to `hostHackathon`, form reset, success toast.
  - Failed submission: error toast shows the message from the thrown error (this form has no retry action, unlike `SubmitProject.js`).

Fixes #10114

## Test plan
- [x] `node --test tests/hostHackathonValidation.test.mjs` passes
- [x] `npm run test:unit` passes (137/137)
- [x] `npx vitest run src/Pages/Hackathons/HostHackathon.test.js` passes (6/6)

### CODE REVIEW

File: src/Pages/Hackathons/HostHackathon.js [hunk 1]
```diff
@@ -67,78 +68,7 @@ const HostHackathon = () => {
...
-    }
-
-    // Participant Limit validation
-    if (data.participantLimit && Number(data.participantLimit) < 1) {
-      newErrors.participantLimit = "Participant limit must be at least 1!";
-    }
-
-    return newErrors;
-  };
+  const requiredFields = REQUIRED_FIELDS;
```
[the remark this answers is not in the document]
Comment (trimoyee-g) [reply src/Pages/Hackathons/HostHackathon.js:71]: This is pre-existing complexity in the component's JSX render (conditional field rendering, `.map()` callbacks, and repeated `isSubmitting`/`prefersReducedMotion` ternaries across the ~500-line return block) — not something introduced by this PR. This diff only *removes* complexity by extracting `validateForm()` into `src/utils/hostHackathonValidation.js`; it's showing up as "new" because this is the first PR to touch the file since CodeScene was enabled.

Per the linked issue (#10114), the scope here is test coverage only — "component under test, no changes expected." Restructuring the render into subcomponents to reduce cyclomatic complexity would be a much larger, behavior-risk-carrying change on an animated form, and is better tracked as its own follow-up refactor rather than folded into a test-coverage PR. Leaving this as-is for now.

**Comment by trimoyee-g** [comment]
Note on the failing **CodeScene Code Health Review** check: it's flagging pre-existing cyclomatic complexity (30, threshold 9) in the `HostHackathon` component's JSX render — conditional field rendering, `.map()` callbacks, and repeated `isSubmitting`/`prefersReducedMotion` ternaries across the ~500-line return block. It's not something this PR introduces; this diff only *removes* complexity by extracting `validateForm()` into `src/utils/hostHackathonValidation.js` (which is itself kept deliberately low-complexity — see the file). It shows up as "new" because this is the first PR to touch `HostHackathon.js` since CodeScene was enabled.

Per #10114, this PR's scope is test coverage only ("component under test, no changes expected"). Restructuring the render into subcomponents to bring complexity under threshold would be a much larger, behavior-risk-carrying change on an animated form and is better suited to its own follow-up refactor issue rather than being folded into a test-coverage PR.

**Comment by trimoyee-g** [comment]
The failing CodeScene check is due to pre-existing JSX complexity in the HostHackathon component's render block, which I have detailed in my previous comment.

Since this PR strictly focuses on adding test coverage as per [#10114](https://github.com/SandeepVashishtha/Eventra/issues/10114) ("component under test, no changes expected"), refactoring the complex UI render tree is outside of this scope and safer to handle in a dedicated follow-up issue.

**Comment by trimoyee-g** [comment]
hey @SandeepVashishtha kindly review this PR and let me know if changes needed. Thanks!
